iOS/Flows/Conversation/Conversation Header/MembersViewController.swift

```diff
@@ -100,24 +109,48 @@ class MembersViewController: DiffableCollectionViewController<MembersCollectionV
         self.dataSource.appendItems([.member(member)], toSection: .members)
     }
 
+    /// A task for scrolling to a specific chat user.
+    private var scrollToUserTask: Task<Void, Never>?
+
     func scroll(to user: ChatUser) {
-        guard let controller = self.conversationController else { return }
+        self.scrollToUserTask?.cancel()
 
-        let member = Member(displayable: AnyHashableDisplayable(user),
-                            conversationController: controller)
-        guard let ip = self.dataSource.indexPath(for: .member(member)) else { return }
+        self.scrollToUserTask = Task { [weak self] in
+            // Wait for the data to finish loading before we try to scroll to a specific user.
+            await self?.loadDataTask?.value
```

Comment:
The problem was that sometimes the datasource wasn't finished loading its data so we couldn't scroll to the correct user.

Tasks fix this problem really well. Just wait for the load data task to finish, then scroll.

Reply:
brilliant